src/modules/booking/containers/booking-preview/booking-preview.tsx

```diff
@@ -0,0 +1,328 @@
+'use client';
+import { useEffect, useState, useMemo, useCallback } from 'react';
+import { CircularProgressbar, buildStyles } from 'react-circular-progressbar';
+import { useParams } from 'next/navigation';
+import dynamic from 'next/dynamic';
+import getDistance from 'geolib/es/getDistance';
+import { createEvent, type EventAttributes } from 'ics';
+import { getHours, getMinutes, format } from 'date-fns';
+// components
+import { Typography } from '@/modules/core/components/typogrpahy';
+import { BookingDetailBox } from '@/modules/booking/components/booking-preview-detail-box/booking-preview-detail-box';
+import { Button } from '@/modules/core/components/button';
+import { Spinner } from '@/modules/core/components/spinner';
+import { UserContactPopup } from '@/modules/user/components/user-contact-popup';
+// hooks
+import { useBoolean } from 'usehooks-ts';
+// utils
+import { trpc } from '@/modules/core/utils/trpc.utils';
+import { formatDuration } from '@/modules/core/utils/time.utils';
+import { onQueryRetry } from '@/modules/core/utils/query-retry.utils';
+import { getAddressGoogleLink } from '@/modules/location/utils/address.utils';
+// style
+import 'react-circular-progressbar/dist/styles.css';
+
+const Map = dynamic(
+  () => import('@/modules/location/components/map').then((res) => res.Map),
+  {
+    ssr: false,
+  }
+);
+
+const MILLISECONDS_IN_MINUTE = 60000;
+const MILLISECONDS_IN_SECOND = 1000;
+const SECONDS_IN_MINUTE = 60;
+const MINUTES_IN_HOUR = 60;
+const HOURS_IN_DAY = 24;
+
+export const BookingPreview = () => {
+  const params = useParams();
+  const code = params.code;
+  const isContactOpen = useBoolean();
+
+  // state
+  const [userLocation, setUserLocation] = useState<{
+    latitude: number;
+    longitude: number;
+  } | null>(null);
+
+  // user location
+  const getUserLocation = () => {
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition((position) => {
+        const { latitude, longitude } = position.coords;
+
+        setUserLocation({ latitude, longitude });
+      });
+    }
+  };
+
+  useEffect(() => {
+    getUserLocation();
+  }, []);
+
+  // queries
+  const bookingDetails = trpc.booking.getByCode.useQuery({
+    code,
+    expand: ['serviceProfessional'],
+  });
+
+  const professional = trpc.professional.get.useQuery(
+    {
+      // @ts-ignore
+      id: bookingDetails.data?.serviceProfessional?.professional?.userId || '',
```

Comment:
@IhorBalan  `professional` object exist in the response but doesn't appear in types